src/zrc/api/tests/test_auth.py

```diff
@@ -0,0 +1,134 @@
+
+            response = do_request(url)
+
+            self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def assertForbiddenWithCorrectScope(
+            self, url: str, scopes: list, method='get',
+            request_kwargs=None, **extra_claims):
+
+        do_request = getattr(self.client, method)
+        request_kwargs = request_kwargs or {}
+
+        jwt = generate_jwt(scopes=scopes, **extra_claims)
+        self.client.credentials(HTTP_AUTHORIZATION=jwt)
+
+        response = do_request(url, **request_kwargs)
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+
+class ZakenCreateTests(AuthCheckMixin, APITestCase):
+
+    def test_cannot_create_zaak_without_correct_scope(self):
+        url = reverse('zaak-list')
+
+        self.assertForbidden(url, method='post')
+
+
+class ZakenReadTests(AuthCheckMixin, APITestCase):
+
+    def test_cannot_read_without_correct_scope(self):
+        urls = [
+            reverse('zaak-list'),
+            reverse('zaak-detail', kwargs={'uuid': 'dummy'}),
+            reverse('status-list'),
+            reverse('status-detail', kwargs={'uuid': 'dummy'}),
+            reverse('zaakobject-list'),
+            reverse('zaakobject-detail', kwargs={'uuid': 'dummy'}),
+        ]
+
+        for url in urls:
+            with self.subTest(url=url):
+                self.assertForbidden(url, method='get')
+
+    def test_zaaktypes_claim(self):
+        """
+        Assert you can only read ZAAKen of the zaaktypes in the claim.
+        """
+        ZaakFactory.create(zaaktype='https://zaaktype.nl/ok')
+        ZaakFactory.create(zaaktype='https://zaaktype.nl/not_ok')
+        url = reverse('zaak-list')
+        jwt = generate_jwt(
+            scopes=[SCOPE_ZAKEN_ALLES_LEZEN],
+            zaaktypes=['https://zaaktype.nl/ok'],
+        )
+        self.client.credentials(HTTP_AUTHORIZATION=jwt)
+
+        response = self.client.get(url, HTTP_ACCEPT_CRS='EPSG:4326')
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]['zaaktype'], 'https://zaaktype.nl/ok')
+
+    def test_zaaktypes_claim_detail(self):
+        """
+        Assert you can only read ZAAKen of the zaaktypes in the claim.
+        """
+        zaak = ZaakFactory.create(zaaktype='https://zaaktype.nl/not_ok')
+        url = reverse('zaak-detail', kwargs={'uuid': zaak.uuid})
+
+        self.assertForbiddenWithCorrectScope(
+            url, [SCOPE_ZAKEN_ALLES_LEZEN],
+            zaaktypes=['https://zaaktype.nl/ok'],
+            request_kwargs={'HTTP_ACCEPT_CRS': 'EPSG:4326'}
+        )
+
+    def test_zaaktypes_wildcard(self):
+        zaak = ZaakFactory.create()
+
+        list_url = reverse('zaak-list')
+        detail_url = reverse('zaak-detail', kwargs={'uuid': zaak.uuid})
+
+        jwt = generate_jwt(
+            scopes=[SCOPE_ZAKEN_ALLES_LEZEN],
+            zaaktypes=['*'],
```

Comment:
Lijkt me goed. Even nadenken over:
De betekenis van een afwezige "zaaktypes" of lege "zaaktypes" is dan totaal geen rechten? Zelfs met alles lezen?

Reply:
Klopt. Explicit is better than implicit (ook voor audit duidelijker).